### DISCUSSION TITLE
All access tokens for my account suddenly broken, including any new ones

### DISCUSSION THREAD

**Discussion by ell1e** [community/community#125648] [Repositories]
### Select Topic Area

Question

### Body

All access tokens for my account seem to be suddenly broken, including any new ones. I always get `remote: Support for password authentication was removed on August 13, 2021` for any git push, I can't access any repositories of mine anymore.

What's up with that, any potential bug on github's end? My apologies if this is a problem on my end, yet I can't figure out what could have possibly changed. The affected token had no expiration date and a token I just set up today has the same issue.

**Comment by MrShadowRIFAT** [comment]
Hi there,

The error message you are seeing, `remote: Support for password authentication was removed on August 13, 2021`, suggests that GitHub is not accepting password authentication for `git push` operations anymore. Instead, you need to use personal access tokens (PATs) for authentication. Here are a few steps to troubleshoot and resolve the issue:

1. **Ensure Correct Token Usage**:
    - Make sure you are using the PAT in place of your password when performing git operations.
    - The command should look something like this:
        ```bash
        git clone https://<username>:<your_token>@github.com/<username>/<repository>.git
        ```

2. **Check Token Scopes**:
    - Verify that your PAT has the correct scopes/permissions. For repository access, ensure it has the `repo` scope.
    - You can manage your PATs and check their scopes here: [GitHub Developer Settings](https://github.com/settings/tokens).

3. **Token Expiry**:
    - Even if the token had no explicit expiration date, GitHub might have revoked it for security reasons. Try generating a new token and ensure it has the necessary scopes.

4. **Updating Remote URLs**:
    - If you have old remote URLs configured, update them to use the PAT. You can update the remote URL using:
        ```bash
        git remote set-url origin https://<username>:<your_token>@github.com/<username>/<repository>.git
        ```

5. **Token in `.netrc` File**:
    - If you prefer not to include the token in the URL, you can store it in the `.netrc` file for automatic usage:
        ```
        machine github.com
        login <username>
        password <your_token>
        ```

6. **SSH Key Authentication**:
    - Alternatively, consider using SSH keys for authentication. You can generate an SSH key and add it to your GitHub account. Instructions can be found here: [Connecting to GitHub with SSH](https://docs.github.com/en/authentication/connecting-to-github-with-ssh).

If after following these steps, the issue persists, it might be worth checking GitHub's status page to ensure there are no ongoing incidents affecting authentication. Additionally, reaching out to GitHub support could provide more personalized assistance.

Let me know if this helps or if you need further assistance!

Best regards,  
[MrShadowRIFAT](https://rifat.website)

**Reply by ell1e** [reply]
This doesn't really seem relevant, I'm already using tokens and I already tried setting up a new one. Thanks for responding nevertheless!

**Comment by ell1e** [comment]
Hm now it works again :thinking:
